fpx-lib/src/api/handlers/spans.rs

```diff
@@ -8,18 +8,15 @@ use http::StatusCode;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 use tracing::error;
+use crate::data::models::TraceId;
 
 #[tracing::instrument(skip_all)]
 pub async fn span_get_handler(
     State(store): State<BoxedStore>,
-    Path((trace_id, span_id)): Path<(String, String)>,
+    Path((trace_id, span_id)): Path<(TraceId, TraceId)>,
```

Comment:
This is a bit confusing that we have `TraceId` for both the `SpanId` and the `TraceId`.

I'm not sure if we should have two types for this or that we should have 1 more generic named type.

Reply:
i say as they have the same functionality we should just rename the type to be more generic, what about `HexEncodedId`?